File: infra/question/passage.js
db = db.getSiblingDB("krakit-question");

if (!db.getCollectionNames().includes("passages")) {
  db.createCollection("passages", {
    validator: {
      $jsonSchema: {
        bsonType: "object",
        required: ["exam_id", "section_id", "passage_text"],
        properties: {
          exam_id: { bsonType: "string" },
          section_id: { bsonType: "string" },
          subject_id: { bsonType: "string" },

          passage_text: { bsonType: "string" },
          topic: { bsonType: "string" },

          is_active: { bsonType: "bool" },
          created_at: { bsonType: "date" }
        }
      }
    }
  });
}

db.passages.createIndex({ exam_id: 1, section_id: 1 });

print("✅ passages ready");

File: services/exam-service/internal/dto/passage.go
package dto

type Passage struct {
	ID          string `json:"id" bson:"_id,omitempty"`
	ExamID      string `json:"exam_id"`
	SectionID   string `json:"section_id"`
	SubjectID   string `json:"subject_id,omitempty"`
	PassageText string `json:"passage_text"`
	Topic       string `json:"topic,omitempty"`
	IsActive    bool   `json:"is_active"`
}
